From cfe76542b375f36382f090d8a4f71f236bd4f3d8 Mon Sep 17 00:00:00 2001
From: goekce <git2022@aydos.de>
Date: Sat, 25 Feb 2023 08:26:54 +0100
Subject: [PATCH] CMake: rename FeatureSummary.cmake to avoid name clashes

According to https://github.com/OGRECave/ogre/issues/2381:

```
The reason is FeatureSummary is a CMake's module, and /usr/lib64/cmake/Qt6Gui/Qt6GuiDependencies.cmake has a line include(FeatureSummary). Since you have FeatureSummary.cmake in your project, it gets loaded instead of /usr/share/cmake/Modules/FeatureSummary.cmake. The solution is simple: just rename FeatureSummary.cmake in your project.
```

Backports the [patch from upstream](https://github.com/OGRECave/ogre/pull/2383/commits/e34ef06bb0d49ce847efe3a80ebe01c9762ea0b1).
---
 rviz_ogre_vendor/CMakeLists.txt              |  1 +
 rviz_ogre_vendor/featuresummary-rename.patch | 14 ++++++++++++++
 2 files changed, 15 insertions(+)
 create mode 100644 rviz_ogre_vendor/featuresummary-rename.patch

diff --git a/rviz_ogre_vendor/CMakeLists.txt b/rviz_ogre_vendor/CMakeLists.txt
index 7893e5099..2ea9ba7d6 100644
--- a/rviz_ogre_vendor/CMakeLists.txt
+++ b/rviz_ogre_vendor/CMakeLists.txt
@@ -193,6 +193,7 @@ macro(build_ogre)
     PATCH_COMMAND
       ${Patch_EXECUTABLE} -p1 -N < ${CMAKE_CURRENT_SOURCE_DIR}/pragma-patch.diff &&
       ${Patch_EXECUTABLE} -p1 -N < ${CMAKE_CURRENT_SOURCE_DIR}/relocatable.patch &&
+      ${Patch_EXECUTABLE} -p1 -N < ${CMAKE_CURRENT_SOURCE_DIR}/featuresummary-rename.patch &&
       ${Patch_EXECUTABLE} -p1 -N < ${CMAKE_CURRENT_SOURCE_DIR}/C4267.patch
     COMMAND
       ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/FindFreetype.cmake ${CMAKE_CURRENT_BINARY_DIR}/ogre-v1.12.10-prefix/src/ogre-v1.12.10/CMake/Packages/FindFreetype.cmake
diff --git a/rviz_ogre_vendor/featuresummary-rename.patch b/rviz_ogre_vendor/featuresummary-rename.patch
new file mode 100644
index 000000000..1fd4bc34f
--- /dev/null
+++ b/rviz_ogre_vendor/featuresummary-rename.patch
@@ -0,0 +1,14 @@
+diff --git a/CMake/FeatureSummary.cmake b/CMake/OgreFeatureSummary.cmake
+similarity index 100%
+rename from CMake/FeatureSummary.cmake
+rename to CMake/OgreFeatureSummary.cmake
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 8583b8882f0..544a7643d68 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -512,4 +512,4 @@ include(Packaging)
+ 
+ 
+ # Show feature summary
+-include(FeatureSummary)
++include(OgreFeatureSummary)
